DietPi-Software | Mosquitto: Fix install on ARMv8

MichaIng opened the pull request:
**Status**: Ready

**Commit list/description**:
+ DietPi-Software | Mosquitto: Do not combine mosquitto_passwd "-c" and "-b" options, which is not supported on versions lower than 1.6.10, i.e. ARMv8 systems up to Buster.
+ DietPi-Software | Mosquitto: Since we do not install the Mosquitto repo on Bullseye (the Mosquitto versions match), no need to have the Bullseye => Buster suite workaround. As the Bullseye repo version is pretty current, we won't install the Mosquitto repo there until it natively supports Bullseye.

MichaIng committed:
v7.2

+ DietPi-Software | Mosquitto: Do not combine mosquitto_passwd "-c" and "-b" options, which is not supported on versions lower than 1.6.10, i.e. ARMv8 systems up to Buster.
+ DietPi-Software | Mosquitto: Since we do not install the Mosquitto repo on Bullseye (the Mosquitto versions match), no need to have the Bullseye => Buster suite workaround. As the Bullseye repo version is pretty current, we won't install the Mosquitto repo there until it natively supports Bullseye.

MichaIng committed:
v7.2

+ CHANGELOG | Mosquitto: Resolved an issue where the install failed on ARMv8 systems

Joulinar commented on dietpi/dietpi-software:
```diff
@@ -11231,7 +11231,8 @@ _EOF_
 				G_EXEC_PRE_FUNC(){ acommand[5]=$GLOBAL_PW; }
-				G_EXEC mosquitto_passwd -c -b /etc/mosquitto/passwd mosquitto "${GLOBAL_PW//?/X}"
+				> /etc/mosquitto/passwd # Pre-create file, required for pre-v1.6.10 (ARMv8 up to Buster): https://github.com/MichaIng/DietPi/issues/4424
+				G_EXEC mosquitto_passwd -b /etc/mosquitto/passwd mosquitto "${GLOBAL_PW//?/X}"
```
Somehow it still fails on submitting the password. even if I use a plain pw directly it's failing

```
┌─────────────────────────────────────────────────┤ DietPi-Software ├──────────────────────────────────────────────────┐
│  - Command: mosquitto_passwd -b /etc/mosquitto/passwd mosquitto demo                                                 │
│  - Exit code: 1                                                                                                      │
│  - DietPi version: v7.2.1 (MichaIng/mosquitto) | HW_MODEL: 4 | HW_ARCH: 3 | DISTRO: 5                                │
│  - Image creator: DietPi Core Team                                                                                   │
│  - Pre-image: Raspberry Pi OS Lite (64-bit)                                                                          │
│  - Error log:                                                                                                        │
│ Error: -b argument given but password file, username or password missing.                                            │
```

The only way to get it working is to use `G_EXEC touch` instead of `>`.

MichaIng replied:
Strange, `touch` and `>` do exactly the same thing.

Joulinar replied:
At least important for me was to us `G_EXEC`

MichaIng replied:
Which command or wrapper is used cannot play a role. Generally it's working:
```
2021-05-27 15:29:00 root@VM-Buster:/tmp# > passwd
2021-05-27 15:29:10 root@VM-Buster:/tmp# mosquitto_passwd -b passwd test test
2021-05-27 15:29:25 root@VM-Buster:/tmp# cat passwd
test:$6$HDbCfDJoUEYbjoBe$XoQw6RzahbakhdfMSZAj+rbeP/jQ4/0rrmHr4NpUWjynUq0rB3KUYei14dAAZtucFUBOJbNl6mqp1KT/Gboo8Q==
```
Probably timing is an issue... but at least on VM (installed the Debian Buster `mosquitto` package, not the one from the Mosquitto repo) it works when appending the commands like `> passwd; mosquitto_passwd -b passwd test test` to eliminate the delay, on tmpfs and as well on root filesystem.

Testing in ARMv8...

MichaIng replied:
Found it. After the `-c` flag has been removed, the wrong command array entry was replaced, replacing the "XXXXX" with the actual password function-internally to avoid printing it to console.

Actually there now was an additional argument, where the error message is misleading:
```
2021-05-27 16:24:23 root@OdroidN2:/tmp# > passwd
2021-05-27 16:24:30 root@OdroidN2:/tmp# mosquitto_passwd -b passwd test test test
Error: -b argument given but password file, username or password missing.
2021-05-27 16:24:40 root@OdroidN2:/tmp# mosquitto_passwd -b passwd test test
2021-05-27 16:25:21 root@OdroidN2:/tmp# cat passwd
test:$6$mZl3ghjsJtblab7X$z+rTcL+WzCkNjrCY2gJeW7iXQBn/eQnrfkhy6rHQI0+O0eQoRn4L683zw7JIEp0MH5qH49eTH8ZPGHa9IJ57Tg==
```

Joulinar replied:
yep works now

MichaIng committed:
v7.2

+ DietPi-Software | Mosquitto: Fix internal password replacement: Array entry 5 has become entry 4 after removing "-c" flag.
